### PR TITLE
INFRANG-4463 - Track log routing usage & Deprecate SignalFx

### PR DESCRIPTION
**Pull request by andruwm**
**Jira:** [INFRANG-4463](https://clever.atlassian.net/browse/INFRANG-4463)

# About
This PR removes SignalFX from the metrics pipeline, and adds some additional pipeline volume metrics. Additionally, some time was spent on performance.

The changes were large due to the fact that SignalFX metrics models were used as the underlying data model. DataDog is the new replacement.

A new metric was added which tags log route volume (count only not actual byte size) but rule name, app and env.

### CODE REVIEW

File: delay.go [hunk 1]
```diff
@@ -1,35 +1,36 @@
...
 	"time"
 )
 
-var maxDelay = 0 * time.Millisecond
-var maxDelayLock = sync.RWMutex{}
-
-func updateMaxDelay(t time.Time) {
-	maxDelayLock.Lock()
-	defer maxDelayLock.Unlock()
+var maxDelay = atomic.Int64{}
```
Comment (andruwm) [inline delay.go:8]: atomic will have better performance for these kinds of operations over a mutex.

File: delay.go [hunk 2]
```diff
@@ -1,35 +1,36 @@
...
-var maxDelay = 0 * time.Millisecond
-var maxDelayLock = sync.RWMutex{}
-
-func updateMaxDelay(t time.Time) {
-	maxDelayLock.Lock()
-	defer maxDelayLock.Unlock()
+var maxDelay int64 = 0
 
+func updateMaxDelay(ts []time.Time) {
+	cur := atomic.LoadInt64(&maxDelay)
```
Comment (andruwm) [inline delay.go:11]: atomic will have better performance than a mutex for these kinds of operations.

File: alerts_consumer.go [hunk 3]
```diff
@@ -275,42 +243,23 @@ func (c *AlertsConsumer) SendBatch(batch [][]byte, tag string) error {
...
-
-		// For fairly recent logs, let SignalFX assign a timestamp on arrival,
-		// instead of the log's actual timestamp.  This ensures datapoints are sent
-		// in-order per timeseries. (If out of order, SignalFX will drop them
-		// silently.)
-		//
-		// For older logs, use their actual timestamp. This may result in some
-		// dropped logs, but should be less important since we care mainly about
-		// accurate alerting on recent data.
-		if isRecent(pts[idx].Timestamp, 30*time.Second) {
```
Comment (andruwm) [inline alerts_consumer.go:294]: I think this is no longer an issue for DD, but someone can correct me if wrong.

Comment (rgarcia) [reply alerts_consumer.go:294]: yeah i think this is ok

**Review by rgarcia — APPROVED** [review]

**Pull request merged**